src/library/cmd_reader.h

```diff
@@ -35,31 +35,10 @@ namespace xlang::cmd
 
             auto last{ options.end() };
 
-            for (C i = 1; i < argc; ++i)
-            {
-                std::string_view arg{ argv[i] };
-
-                if (arg[0] == '-')
-                {
-                    arg.remove_prefix(1);
-                    last = find(options, arg);
-
-                    if (last == options.end())
-                    {
-                        throw_invalid("Option '-", arg, "' is not supported");
-                    }
-
-                    m_options.try_emplace(last->name);
-                }
-                else if (last == options.end())
-                {
-                    throw_invalid("Value '", arg, "' is not supported");
-                }
-                else
-                {
-                    m_options[last->name].push_back(std::string{ arg });
-                }
-            }
+			for (C i = 1; i < argc; ++i)
+			{
+				extract_option(std::string_view{ std::string{ argv[i] } }, options, last);
```

Comment:
All of these conversions should be unnecessary as the `extract_option` function takes a `string_view`. You should be able to simply pass the `argv[i]` as the first argument.